### PR TITLE
improve date validation

### PR DESCRIPTION
**Pull request by elielijah321**
**What is the change?**
Validate date

**Why do we need the change?**
Requirement

**What is the impact?**
Minimal

**Azure DevOps Ticket**
107050

### CODE REVIEW

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Decision/Add.cshtml.cs [hunk 1]
```diff
@@ -127,8 +132,13 @@ private void PopulateCreateDecisionDtoFromRequest()
 			var dtr_month = Request.Form["dtr-month-request-received"];
 			var dtr_year = Request.Form["dtr-year-request-received"];
 			var dtString = $"{dtr_day}-{dtr_month}-{dtr_year}";
-			var dateStarted = DateTimeHelper.TryParseExact(dtString, out DateTime parsedDate) ? parsedDate : (DateTime?)null;
-						
+			var isValidDate = DateTimeHelper.TryParseExact(dtString, out DateTime parsedDate);
+
+			if (dtString != "--" && !isValidDate)
+			{
+				throw new InvalidOperationException($"{dtString} is an invaid date");
```
Comment (emma-w-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Decision/Add.cshtml.cs:139]: I think this shouldn't be an invalidoperationexception - https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-7.0
"[InvalidOperationException](https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-7.0) is used in cases when the failure to invoke a method is caused by reasons other than invalid arguments. Typically, it is thrown when the state of an object cannot support the method call."

Actually I can see we've used this before, so maybe one to revisit later & update globally (we ought to have our own application exceptions really)

Comment (emma-w-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Decision/Add.cshtml.cs:139]: spelling! "invaid"--> "invalid"

File: ConcernsCaseWork/ConcernsCaseWork.Tests/Pages/Case/Management/Action/Decision/AddPageModelTests.cs [hunk 2]
```diff
@@ -163,6 +163,38 @@ public async Task OnPostAsync_When_Validation_Failures_Concatenates_Error_Messag
...
 
+		[Test]
+		public async Task OnPostAsync_When_InvalidDate_Error_Message()
+		{
+			long caseUrn = 233433;
+			var invalidDay = "32";
+			var invalidMonth = "13";
+			var invalidYear = "7";
+
+			var expectedErrorMessage = $"{invalidDay}-{invalidMonth}-{invalidYear} is an invaid date";
```
Comment (emma-w-nimble) [inline ConcernsCaseWork/ConcernsCaseWork.Tests/Pages/Case/Management/Action/Decision/AddPageModelTests.cs:174]: spelling

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Decision/Add.cshtml.cs [hunk 3]
```diff
@@ -127,8 +132,13 @@ private void PopulateCreateDecisionDtoFromRequest()
 			var dtr_month = Request.Form["dtr-month-request-received"];
 			var dtr_year = Request.Form["dtr-year-request-received"];
 			var dtString = $"{dtr_day}-{dtr_month}-{dtr_year}";
-			var dateStarted = DateTimeHelper.TryParseExact(dtString, out DateTime parsedDate) ? parsedDate : (DateTime?)null;
-						
+			var isValidDate = DateTimeHelper.TryParseExact(dtString, out DateTime parsedDate);
```
Comment (emma-w-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Decision/Add.cshtml.cs:135]: Should we be using a culture here - does this work regardless of the local machine settings (US date format / UK date format)?

**Review by emma-w-nimble — COMMENTED** [review]
Hiya, I've left a couple of comments. One spelling which needs correcting, a question, and an observation :)

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Decision/Add.cshtml.cs (hunk 1 above)
Comment (elielijah321) [reply ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Decision/Add.cshtml.cs]: Will be discussed in the tech debt meeting

**Pull request merged**